http_job.go

```diff
@@ -80,10 +81,12 @@ func (j *httpJob) Error(ctx gocontext.Context, errMessage string) error {
 		return err
 	}
 
-	return j.Finish(FinishStateErrored)
+	return j.Finish(ctx, FinishStateErrored)
 }
 
-func (j *httpJob) Requeue() error {
+func (j *httpJob) Requeue(ctx gocontext.Context) error {
+	context.LoggerFromContext(ctx).WithField("job", j.Payload().Job.ID).Info("requeueing job")
```

Comment:
I don't think we are currently setting it on the context. I wasn't able to find any reference to `context.WithValue()`. I agree that we should do that though, that would clean up a whole lot of code.